style: apply /style-guide pass to weave (#2739)
wandb/docs: weave/guides/tracking/trace-disable.mdx

---
title: "Disable tracing"
description: "Learn options to disable or conditionally turn off W&B Weave tracing"
keywords: ["WEAVE_DISABLED", "tracing_disabled", "context manager", "disabled flag"]
---

This page describes how to disable W&B Weave tracing in your application. Use these options when you want to prevent traces from being logged, for example, during local development, in test environments, or for specific code paths where tracing isn't needed. Choose the approach that best matches the scope of control you need: program-wide, per-client, or per-code-block.

## Environment variable

Use an environment variable when you want to disable tracing for the entire program without modifying application code.

To unconditionally disable tracing for the entire program, set the environment variable `WEAVE_DISABLED=true`.

`WEAVE_DISABLED` is read only once, at function-definition time. You can't use this variable to toggle tracing at runtime.

## Client initialization

> Use the client `disabled` setting when you want to control tracing based on runtime conditions at the point where you initialize the Weave client.

Reviewer: ```suggestion
```

Author: Removed the redundant intro sentence in 23750a6f6 (per the empty suggestion).

> To conditionally enable tracing for a specific initialization based on a runtime condition, initialize the client with the `disabled` flag in init settings.

Reviewer: ```suggestion
To conditionally disable tracing for a specific Weave client, initialize the client with the `disabled` flag in init settings.
```

Author: Applied in 23750a6f6 (also fixes the prior "enable" -> "disable" on this disable page).

## Context manager

> Use the context manager when you want fine-grained control over tracing within a specific block of code, while leaving tracing enabled elsewhere in your application.

Reviewer: ```suggestion
```

Author: Removed the redundant intro sentence in ff53ef96e (per the empty suggestion).

> To conditionally disable tracing for a specific block of code, use a tracing context manager. Use `with tracing_disabled()` to suppress tracing **only for the function calls executed inside the `with` block**. Use it in application code to scope which calls shouldn't be logged.

Reviewer: ```suggestion
To conditionally disable tracing only within a specific block of code, use a tracing context manager. Use `with tracing_disabled()` to suppress tracing **only for the function calls executed inside the `with` block**. Use it in application code to scope which calls shouldn't be logged.
```

Author: Applied verbatim in ff53ef96e.
